secure-api-gateway-ig-extensions/src/main/java/com/forgerock/sapi/gateway/dcr/utils/JwtReconstructionException.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright © 2020-2022 ForgeRock AS (obst@forgerock.com)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.forgerock.sapi.gateway.dcr.utils;
+
+/**
+ * Checked exception that can be thrown when reconstruction of a JWT from a b64 encoded JWT string fails
+ */
+public class JwtReconstructionException extends Exception {
```

Comment:
The jose JwtReconstruction is a Runtime exception. 
I want this to be a checked exception so that I can know when the user has supplied a badly formatted jwt and write a good error about it. Unchecked makes sense if it's a config thing and affects system start up, e.g. the jwt is a part of our configuration, but when user supplied it's important to know when the third party library throws an exception. 

I think for our system a utility method that makes you have to handle the exception makes more sense.